src/helpers/glpTEST/getGlpSwapData.js

```diff
@@ -0,0 +1,319 @@
+import { ethers, providers } from "ethers";
+import { swap0xRequest } from "@/helpers/0x";
+import { actions } from "@/helpers/cauldron/cook/actions";
+
+import gmxVaultAbi from "@/helpers/glpTEST/abi/gmxVault";
+import gmxLensAbi from "@/helpers/glpTEST/abi/gmxLens";
+import { priceAbi } from "@/utils/farmPools/abi/priceAbi";
+const gmxVaultAddress = "0x489ee077994B6658eAfA855C308275EAd8097C4A";
+const gmxLensAddress = "0x9f8fB63ef774A496cd9aD3Cc29c1e81Ae947FC2e";
+import store from "@/store";
+
+const blacklis = [
+  "0x17fc002b466eec40dae837fc4be5c67993ddbd6f",
+  "0xfea7a6a0b346362bf88a9e4a88416b77a57d6c2a",
+  "0xfd086bc7cd5c481dcc9c85ebe478a1c0b69fcbb9",
+  "0xda10009cbd5d07dd0cecc66161fc93d7c9000da1"
+];
+
+const getMimOraclePrice = async () => {
+  try {
+    const defaultProvider = new providers.StaticJsonRpcProvider(
+      "https://mainnet.infura.io/v3/9aa3d95b3bc440fa88ea12eaa4456161"
+    );
+
+    const oracle = new ethers.Contract(
+      "0x7A364e8770418566e3eb2001A96116E6138Eb32F",
+      JSON.stringify(priceAbi),
+      defaultProvider
+    );
+
+    return await oracle.latestAnswer();
+  } catch (error) {
+    console.log("getMimOraclePrice err:", error);
+  }
+};
+
+const getNameFromAddress = (address) => {
+  return {
+    "0x2f2a2543B76A4166549F7aaB2e75Bef0aefC5B0f": "WBTC",
+    "0x82aF49447D8a07e3bd95BD0d56f35241523fBab1": "WETH",
+    "0xFF970A61A04b1cA14834A43f5dE4533eBDDB5CC8": "USDC",
+    "0xf97f4df75117a78c1A5a0DBb814Af92458539FB4": "LINK",
+    "0xFa7F8980b0f1E64A2062791cc3b0871572f1F7f0": "UNI",
+    "0xFd086bC7CD5C481DCC9C85ebE478A1C0b69FCbb9": "USDT",
+    "0xFEa7a6a0B346362BF88A9e4A88416B77a57D6c2A": "MIM",
+    "0x17FC002b466eEc40DaE837Fc4bE5c67993ddBd6F": "FRAX",
+    "0xDA10009cBd5D07dd0CeCc66161FC93D7c9000da1": "DAI",
+  }[address];
+};
+
+const maxGlpAmount = (a, b) => {
+  return +b.glpOutAmount - +a.glpOutAmount;
+};
+
+const maxBuyAmount = (a, b) => {
+  return +a.buyAmount > +b.buyAmount ? a : b;
+};
+
+const getTokens = async (provider, lensContract) => {
+  const mimOraclePrice = await getMimOraclePrice();
+
+  const gmxVaultContract = await new ethers.Contract(
+    gmxVaultAddress,
+    JSON.stringify(gmxVaultAbi),
+    provider
+  );
+
+  const whitelistedTokensLength =
+    await gmxVaultContract.allWhitelistedTokensLength();
+
+  const tokens = [];
+  for (let i = 0; i < whitelistedTokensLength; i++) {
+    const address = await gmxVaultContract.allWhitelistedTokens(i);
+    // Exclude FRAX and MIM.
+    if (blacklis.indexOf(address.toLowerCase()) === -1) {
+      const maxAmountIn = await lensContract.getMaxAmountIn(address);
+      const maxAmountInToMIM = maxAmountIn.div(mimOraclePrice).mul(1e8);
```

Comment:
not necessary, without this conversion we just allocate slightly less than potentially possible